### PR TITLE
improve code

### LINKED ISSUE #3: CVE-2020-15366 (Medium) detected in ajv-6.11.0.tgz
## CVE-2020-15366 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>ajv-6.11.0.tgz</b></p></summary>

<p>Another JSON Schema Validator</p>
<p>Library home page: <a href="https://registry.npmjs.org/ajv/-/ajv-6.11.0.tgz">https://registry.npmjs.org/ajv/-/ajv-6.11.0.tgz</a></p>
<p>Path to dependency file: mwengeSMS/package.json</p>
<p>Path to vulnerable library: mwengeSMS/node_modules/ajv/package.json</p>
<p>

Dependency Hierarchy:
  - africastalking-0.4.5.tgz (Root Library)
    - unirest-0.6.0.tgz
      - request-2.88.2.tgz
        - har-validator-5.1.3.tgz
          - :x: **ajv-6.11.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/Kijacode/mwengeSMS/commit/a0d7a6bf7e5a67a68d28acad4d0f2d09abcbf6e0">a0d7a6bf7e5a67a68d28acad4d0f2d09abcbf6e0</a></p>
<p>Found in base branch: <b>master</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
An issue was discovered in ajv.validate() in Ajv (aka Another JSON Schema Validator) 6.12.2. A carefully crafted JSON schema could be provided that allows execution of other code by prototype pollution. (While untrusted schemas are recommended against, the worst case of an untrusted schema should be a denial of service, not execution of code.)

<p>Publish Date: 2020-07-15
<p>URL: <a href=https://vuln.whitesourcesoftware.com/vulnerability/CVE-2020-15366>CVE-2020-15366</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>5.6</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: High
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: Low
  - Integrity Impact: Low
  - Availability Impact: Low
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Upgrade version</p>
<p>Origin: <a href="https://github.com/ajv-validator/ajv/releases/tag/v6.12.3">https://github.com/ajv-validator/ajv/releases/tag/v6.12.3</a></p>
<p>Release Date: 2020-07-15</p>
<p>Fix Resolution: ajv - 6.12.3</p>

</p>
</details>
<p></p>

***
Step up your Open Source Security Game with WhiteSource [here](https://www.whitesourcesoftware.com/full_solution_bolt_github)

### LINKED ISSUE #4: CVE-2020-7598 (Medium) detected in minimist-0.0.8.tgz, minimist-1.2.0.tgz
## CVE-2020-7598 - Medium Severity Vulnerability
<details><summary> Vulnerable Libraries - <b>minimist-0.0.8.tgz</b>, <b>minimist-1.2.0.tgz</b></p></summary>
<p>

<details><summary><b>minimist-0.0.8.tgz</b></p></summary>

<p>parse argument options</p>
<p>Library home page: <a href="https://registry.npmjs.org/minimist/-/minimist-0.0.8.tgz">https://registry.npmjs.org/minimist/-/minimist-0.0.8.tgz</a></p>
<p>Path to dependency file: mwengeSMS/package.json</p>
<p>Path to vulnerable library: mwengeSMS/node_modules/grpc/node_modules/mkdirp/node_modules/minimist/package.json</p>
<p>

Dependency Hierarchy:
  - africastalking-0.4.5.tgz (Root Library)
    - grpc-1.24.2.tgz
      - node-pre-gyp-0.14.0.tgz
        - mkdirp-0.5.1.tgz
          - :x: **minimist-0.0.8.tgz** (Vulnerable Library)
</details>
<details><summary><b>minimist-1.2.0.tgz</b></p></summary>

<p>parse argument options</p>
<p>Library home page: <a href="https://registry.npmjs.org/minimist/-/minimist-1.2.0.tgz">https://registry.npmjs.org/minimist/-/minimist-1.2.0.tgz</a></p>
<p>Path to dependency file: mwengeSMS/package.json</p>
<p>Path to vulnerable library: mwengeSMS/node_modules/grpc/node_modules/minimist/package.json</p>
<p>

Dependency Hierarchy:
  - africastalking-0.4.5.tgz (Root Library)
    - grpc-1.24.2.tgz
      - node-pre-gyp-0.14.0.tgz
        - rc-1.2.8.tgz
          - :x: **minimist-1.2.0.tgz** (Vulnerable Library)
</details>

<p>Found in HEAD commit: <a href="https://github.com/Kijacode/mwengeSMS/commit/a0d7a6bf7e5a67a68d28acad4d0f2d09abcbf6e0">a0d7a6bf7e5a67a68d28acad4d0f2d09abcbf6e0</a></p>
<p>Found in base branch: <b>master</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
minimist before 1.2.2 could be tricked into adding or modifying properties of Object.prototype using a "constructor" or "__proto__" payload.

<p>Publish Date: 2020-03-11
<p>URL: <a href=https://vuln.whitesourcesoftware.com/vulnerability/CVE-2020-7598>CVE-2020-7598</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>5.6</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: High
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: Low
  - Integrity Impact: Low
  - Availability Impact: Low
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Upgrade version</p>
<p>Origin: <a href="https://github.com/substack/minimist/commit/63e7ed05aa4b1889ec2f3b196426db4500cbda94">https://github.com/substack/minimist/commit/63e7ed05aa4b1889ec2f3b196426db4500cbda94</a></p>
<p>Release Date: 2020-03-11</p>
<p>Fix Resolution: minimist - 0.2.1,1.2.3</p>

</p>
</details>
<p></p>

***
Step up your Open Source Security Game with WhiteSource [here](https://www.whitesourcesoftware.com/full_solution_bolt_github)

### LINKED ISSUE #8: CVE-2020-7768 (High) detected in grpc-1.24.2.tgz
## CVE-2020-7768 - High Severity Vulnerability
<details><summary> Vulnerable Library - <b>grpc-1.24.2.tgz</b></p></summary>

<p>gRPC Library for Node</p>
<p>Library home page: <a href="https://registry.npmjs.org/grpc/-/grpc-1.24.2.tgz">https://registry.npmjs.org/grpc/-/grpc-1.24.2.tgz</a></p>
<p>Path to dependency file: mwengeSMS/package.json</p>
<p>Path to vulnerable library: mwengeSMS/node_modules/grpc/package.json</p>
<p>

Dependency Hierarchy:
  - africastalking-0.4.5.tgz (Root Library)
    - :x: **grpc-1.24.2.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/Kijacode/mwengeSMS/commit/51402e4003b52f720682d8da0a1ccee128508385">51402e4003b52f720682d8da0a1ccee128508385</a></p>
<p>Found in base branch: <b>master</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
The package grpc before 1.24.4; the package @grpc/grpc-js before 1.1.8 are vulnerable to Prototype Pollution via loadPackageDefinition.

<p>Publish Date: 2020-11-11
<p>URL: <a href=https://vuln.whitesourcesoftware.com/vulnerability/CVE-2020-7768>CVE-2020-7768</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>7.5</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: None
  - Integrity Impact: None
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Upgrade version</p>
<p>Origin: <a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7768">https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7768</a></p>
<p>Release Date: 2020-07-21</p>
<p>Fix Resolution: grpc 1.24.4, grpc-js 1.1.8</p>

</p>
</details>
<p></p>

***
Step up your Open Source Security Game with WhiteSource [here](https://www.whitesourcesoftware.com/full_solution_bolt_github)

### LINKED ISSUE #9: CVE-2020-7788 (High) detected in ini-1.3.5.tgz
## CVE-2020-7788 - High Severity Vulnerability
<details><summary> Vulnerable Library - <b>ini-1.3.5.tgz</b></p></summary>

<p>An ini encoder/decoder for node</p>
<p>Library home page: <a href="https://registry.npmjs.org/ini/-/ini-1.3.5.tgz">https://registry.npmjs.org/ini/-/ini-1.3.5.tgz</a></p>
<p>Path to dependency file: mwengeSMS/package.json</p>
<p>Path to vulnerable library: mwengeSMS/node_modules/grpc/node_modules/ini/package.json</p>
<p>

Dependency Hierarchy:
  - africastalking-0.4.5.tgz (Root Library)
    - grpc-1.24.2.tgz
      - node-pre-gyp-0.14.0.tgz
        - rc-1.2.8.tgz
          - :x: **ini-1.3.5.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/Kijacode/mwengeSMS/commit/51402e4003b52f720682d8da0a1ccee128508385">51402e4003b52f720682d8da0a1ccee128508385</a></p>
<p>Found in base branch: <b>master</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
This affects the package ini before 1.3.6. If an attacker submits a malicious INI file to an application that parses it with ini.parse, they will pollute the prototype on the application. This can be exploited further depending on the context.

<p>Publish Date: 2020-12-11
<p>URL: <a href=https://vuln.whitesourcesoftware.com/vulnerability/CVE-2020-7788>CVE-2020-7788</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>7.3</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: Low
  - Integrity Impact: Low
  - Availability Impact: Low
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Upgrade version</p>
<p>Origin: <a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7788">https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7788</a></p>
<p>Release Date: 2020-12-11</p>
<p>Fix Resolution: v1.3.6</p>

</p>
</details>
<p></p>

***
Step up your Open Source Security Game with WhiteSource [here](https://www.whitesourcesoftware.com/full_solution_bolt_github)

### LINKED ISSUE #10: CVE-2020-28168 (Medium) detected in axios-0.19.2.tgz
## CVE-2020-28168 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>axios-0.19.2.tgz</b></p></summary>

<p>Promise based HTTP client for the browser and node.js</p>
<p>Library home page: <a href="https://registry.npmjs.org/axios/-/axios-0.19.2.tgz">https://registry.npmjs.org/axios/-/axios-0.19.2.tgz</a></p>
<p>Path to dependency file: mwengeSMS/package.json</p>
<p>Path to vulnerable library: mwengeSMS/node_modules/africastalking/node_modules/axios/package.json</p>
<p>

Dependency Hierarchy:
  - africastalking-0.4.5.tgz (Root Library)
    - :x: **axios-0.19.2.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/Kijacode/mwengeSMS/commit/51402e4003b52f720682d8da0a1ccee128508385">51402e4003b52f720682d8da0a1ccee128508385</a></p>
<p>Found in base branch: <b>master</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
Axios NPM package 0.21.0 contains a Server-Side Request Forgery (SSRF) vulnerability where an attacker is able to bypass a proxy by providing a URL that responds with a redirect to a restricted host or IP address.

<p>Publish Date: 2020-11-06
<p>URL: <a href=https://vuln.whitesourcesoftware.com/vulnerability/CVE-2020-28168>CVE-2020-28168</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>5.9</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: High
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: High
  - Integrity Impact: None
  - Availability Impact: None
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>

***
Step up your Open Source Security Game with WhiteSource [here](https://www.whitesourcesoftware.com/full_solution_bolt_github)

### LINKED ISSUE #11: CVE-2020-7774 (High) detected in y18n-3.2.1.tgz
## CVE-2020-7774 - High Severity Vulnerability
<details><summary> Vulnerable Library - <b>y18n-3.2.1.tgz</b></p></summary>

<p>the bare-bones internationalization library used by yargs</p>
<p>Library home page: <a href="https://registry.npmjs.org/y18n/-/y18n-3.2.1.tgz">https://registry.npmjs.org/y18n/-/y18n-3.2.1.tgz</a></p>
<p>Path to dependency file: mwengeSMS/package.json</p>
<p>Path to vulnerable library: mwengeSMS/node_modules/y18n/package.json</p>
<p>

Dependency Hierarchy:
  - africastalking-0.4.5.tgz (Root Library)
    - grpc-1.24.2.tgz
      - protobufjs-5.0.3.tgz
        - yargs-3.32.0.tgz
          - :x: **y18n-3.2.1.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/Kijacode/mwengeSMS/commit/51402e4003b52f720682d8da0a1ccee128508385">51402e4003b52f720682d8da0a1ccee128508385</a></p>
<p>Found in base branch: <b>master</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
This affects the package y18n before 3.2.2, 4.0.1 and 5.0.5. PoC by po6ix: const y18n = require('y18n')(); y18n.setLocale('__proto__'); y18n.updateLocale({polluted: true}); console.log(polluted); // true

<p>Publish Date: 2020-11-17
<p>URL: <a href=https://vuln.whitesourcesoftware.com/vulnerability/CVE-2020-7774>CVE-2020-7774</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>7.3</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: Low
  - Integrity Impact: Low
  - Availability Impact: Low
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Upgrade version</p>
<p>Origin: <a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7774">https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7774</a></p>
<p>Release Date: 2020-11-17</p>
<p>Fix Resolution: 5.0.5</p>

</p>
</details>
<p></p>

***
Step up your Open Source Security Game with WhiteSource [here](https://www.whitesourcesoftware.com/full_solution_bolt_github)

### PR DESCRIPTION
**Pull request by gsonly**
## TODO

- [x] rm extra deps
- [ ] format code 
- [ ] fix typos

fixes: #7 :wink: 

closes #11 
closes #10
closes #9
closes #8
closes #3
closes #4

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SECRET_KEY=
+API_KEY=
```
Comment (alphaolomi) [inline .env.example:2]: Environment variables are better than js objects

File: app/credentiials/credential.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  appname: 'sandbox',
+  key: '',
+  Bongolive_apikey: ' ',
+  Bongolive_username: '',
+  Bongolive_password: '',
+  Bongolive_sendername: '',
+  Bongolive_phonenumber: '',
```
Comment (alphaolomi) [inline app/credentiials/credential.js:8]: Hide sensitive data

File: app/helpers/sms.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+const axios = require("axios");
```
Comment (alphaolomi) [inline app/helpers/sms.js:1]: fixed this 

it works 💯

File: app/routes/api.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const { Router } = require("express");
+const { bongoSendSms } = require("../helpers/sms");
+
+const router = Router();
+
+router.get("/", (req, res) => {
```
Comment (alphaolomi) [inline app/routes/api.js:6]: added simple route for app health checking

File: package.json [hunk 5]
```diff
@@ -1,18 +1,16 @@
...
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "author": "Kijacode",
   "license": "ISC",
   "dependencies": {
-    "africastalking": "^0.4.5",
+    "africastalking": "^0.5.2",
     "axios": "^0.21.1",
-    "body-parser": "^1.19.0",
-    "btoa": "^1.2.1",
```
Comment (alphaolomi) [inline package.json:15]: remove this in favor of Buffer which in built-in node env,

**Review by alphaolomi — APPROVED** [review]

**Comment by alphaolomi** [comment]
@gsonly 

> [ ] format code

Code formatting should be automated recommending use of prettier for this

**Comment by gsonly** [comment]
Shall we enforce a style guide for linting and formatting?

**Comment by alphaolomi** [comment]
Yes a project wide style guide would be nice

File: .env.example (hunk 1 above)
Comment (Kijacode) [reply .env.example:2]: @alphaolomi  okay,let me clean this up and add environment variables

**Pull request merged**